src/osip-monitoring-controller/src/main/java/edu/kit/pse/osip/monitoring/controller/MonitoringController.java

```diff
@@ -82,12 +124,498 @@ private void setupUI(Stage stage) {
+            public void onReceived(boolean value) {
+                try {
+                    client.subscribeTemperature(-1,
+                            immValue -> tank.setLiquid(new Liquid(tank.getLiquid().getAmount(), immValue,
+                                    tank.getLiquid().getColor())));
+                } catch (UAClientException e) {
+                    System.err.println(e.getMessage());
+                }
+            }
+        };
+        BooleanReceivedListener underflowSensorListener = new BooleanReceivedListener() {
+            public void onReceived(boolean value) {
+                try {
+                    client.subscribeFillLevel(-1,
+                            immValue -> tank.setLiquid(new Liquid(immValue * tank.getCapacity(),
+                                    tank.getLiquid().getTemperature(), tank.getLiquid().getColor())));
+                } catch (UAClientException e) {
+                    System.err.println(e.getMessage());
+                }
+            }
+        };   
+        ErrorListener errorListener = new ErrorListener() {
+            public void onError(int code) {
+                switch(code) {
+                    case 1: System.err.println("The client disconnected from the server!");
+                            break;
+                    case 2: System.err.println("The data type found on the server does not match the one you requested!");
+                            break;
+                    case 3: System.err.println("The server returned an unsupported data type");
+                            break;
+                }
+            }
+        };
+    }
+    
+    private class MixTankContainer {
+        private MixTank tank;
+        private MixTankClient client;
+        IntReceivedListener colorListener = new IntReceivedListener() {
+            public void onReceived(int value) {
+                tank.setLiquid(new Liquid(tank.getLiquid().getAmount(), tank.getLiquid().getTemperature(), new Color(value)));
+            }
+        };
+        FloatReceivedListener fillLevelListener = new FloatReceivedListener() {
+            public void onReceived(float value) {
+                tank.setLiquid(new Liquid(value * tank.getCapacity(), tank.getLiquid().getTemperature(), tank.getLiquid().getColor()));
+            }
+        };
+        IntReceivedListener motorSpeedListener = new IntReceivedListener() {
+            public void onReceived(int value) {
+                mixCont.tank.getMotor().setRPM(value);
+            }
+        };
+        FloatReceivedListener outputFlowRateListener = new FloatReceivedListener() {
+            public void onReceived(float value) {
+                tank.getOutPipe().setValveThreshold((byte)(value * 100));
+            }
+        };
+        BooleanReceivedListener overflowSensorListener = new BooleanReceivedListener() {
+            public void onReceived(boolean value) {
+                try {
+                    client.subscribeFillLevel(-1,
```

Comment:
Da wäre es vielleicht noch sinnvoll, eine Konstante in meine Klasse einzubauen. Sonst sieht die -1 seltsam aus. Besser: UAClientWrapper.SINGLE_READ oder so